Include ViewPanel.Top/Left in E2E tree dumps

rectified95 opened the pull request:
Including  Top/Left for tests which dump a ViewPanel.
Closes #7129

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7279)

NickGerleman commented on packages/e2e-test-app/test/LegacyImageTest.test.ts:
```diff
@@ -14,27 +14,35 @@ beforeAll(async () => {
     await toggleRTLMode();
-    const dump = await dumpVisualTree('image-container');
+    const dump = await dumpVisualTree('image-container', {
+      additionalProperties: ['Left', 'Top'],
```
Instead of updating each of rhe callsites, could we do this in the library function? That way it is propagated everywhere.

rectified95 replied:
It's not possible to enumerate attached properties in XAML, and the lib is used by other projects as well, so it's necessary to pass a list of attached props you want it to extract via reflection.

NickGerleman replied:
Can we pass this in dumpVisualTree?

rectified95 replied:
Passing this now via the entry point to the test app.

NickGerleman commented:
@rectified95 Are you sure we are correctly capturing the ViewPanel attached properties?

I would expect us see the properties in more places in the dump, since they are present on every RN view. I would also expect to see some non-zero values, since these props will show up for anything participating in Flexbox layout.

rectified95 commented:
@NickGerleman Good point about all the values being zero, will look into it.
About the number of instances - I passed Left/Top for all tests whose dumps included a ViewPanel entry, and the new snapshots include Top/Left values for ViewPanels we currently dump.

Also, I don't wanna spend too much time on this - turns out there's another task to remove the attached Left/Top altogether...
https://github.com/microsoft/react-native-windows/issues/6634

NickGerleman commented:
> I passed Left/Top for all tests whose dumps included a ViewPanel entry

The naming here is confusing, but ViewPanel attached properties aren't set on ViewPanels. They're set on anything with a shadow node. So every native component should have them.

The way this works is that there is a ViewPanel created above every shadownode. The ViewPanel customizes its arrange pass to position child elements according to their ViewPanel attached properties.

rectified95 commented:
@asklar Can you comment on the above?

NickGerleman commented:
This one should be easy enough to validate by opening the live visual tree in Visual Studio (with "Just My XAML" disabled).

asklar commented:
It looks like Left/Top is non-zero for some elements, here's the dump I got:
<detail>
```
{
  "XamlType": "ReactUWPTestApp.MainPage",
  "Background": "#FFFFFFFF",
  "BorderBrush": null,
  "BorderThickness": "0,0,0,0",
  "Clip": null,
  "CornerRadius": "0,0,0,0",
  "FlowDirection": "LeftToRight",
  "Foreground": "#FF000000",
  "HorizontalAlignment": "Stretch",
  "Margin": "0,0,0,0",
  "Padding": "0,0,0,0",
  "RenderSize": [800, 600],
  "VerticalAlignment": "Stretch",
  "Visibility": "Visible",
  "children": [
  {
    "XamlType": "Microsoft.ReactNative.ReactRootView",
    "Background": "#FFFFFFFF",
    "BorderBrush": null,
    "BorderThickness": "0,0,0,0",
    "Clip": null,
    "CornerRadius": "0,0,0,0",
    "FlowDirection": "LeftToRight",
    "Height": 600,
    "HorizontalAlignment": "Stretch",
    "Margin": "0,0,0,0",
    "Name": "myRootView",
    "Padding": "0,0,0,0",
    "RenderSize": [800, 600],
    "VerticalAlignment": "Stretch",
    "Visibility": "Visible",
    "children": [
    {
      "XamlType": "Windows.UI.Xaml.Controls.Grid",
      "Background": null,
      "BorderBrush": null,
      "BorderThickness": "0,0,0,0",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "HorizontalAlignment": "Stretch",
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "RenderSize": [800, 600],
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "children": [
      {
        "XamlType": "Microsoft.ReactNative.ViewPanel",
        "Background": null,
        "BorderBrush": "#00000000",
        "BorderThickness": "0,0,0,0",
        "Clip": null,
        "CornerRadius": "0,0,0,0",
        "FlowDirection": "LeftToRight",
        "Height": 600,
        "HorizontalAlignment": "Stretch",
        "Margin": "0,0,0,0",
        "RenderSize": [800, 600],
        "VerticalAlignment": "Stretch",
        "Visibility": "Visible",
        "Width": 800,
        "Left": 0,
        "Top": 0,
        "children": [
        {
          "XamlType": "Microsoft.ReactNative.ViewPanel",
          "Background": null,
          "BorderBrush": "#00000000",
          "BorderThickness": "0,0,0,0",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "Height": 600,
          "HorizontalAlignment": "Stretch",
          "Margin": "0,0,0,0",
          "RenderSize": [800, 600],
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "Width": 800,
          "Left": 0,
          "Top": 0,
          "children": [
          {
            "XamlType": "Microsoft.ReactNative.ViewPanel",
            "Background": null,
            "BorderBrush": "#00000000",
            "BorderThickness": "0,0,0,0",
            "Clip": null,
            "CornerRadius": "0,0,0,0",
            "FlowDirection": "LeftToRight",
            "Height": 600,
            "HorizontalAlignment": "Stretch",
            "Margin": "0,0,0,0",
            "RenderSize": [800, 600],
            "VerticalAlignment": "Stretch",
            "Visibility": "Visible",
            "Width": 800,
            "Left": 0,
            "Top": 0,
            "children": [
            {
              "XamlType": "Microsoft.ReactNative.ViewPanel",
              "Background": null,
              "BorderBrush": "#00000000",
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "Height": 56,
              "HorizontalAlignment": "Stretch",
              "Margin": "0,0,0,0",
              "RenderSize": [800, 56],
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "Width": 800,
              "Left": 0,
              "Top": 0,
              "children": [
              {
                "XamlType": "Microsoft.ReactNative.ViewPanel",
                "Background": "#FFF3F8FF",
                "BorderBrush": "#00000000",
                "BorderThickness": "0,0,0,0",
                "Clip": null,
                "CornerRadius": "0,0,0,0",
                "FlowDirection": "LeftToRight",
                "Height": 56,
                "HorizontalAlignment": "Stretch",
                "Margin": "0,0,0,0",
                "RenderSize": [800, 56],
                "VerticalAlignment": "Stretch",
                "Visibility": "Visible",
                "Width": 800,
                "Left": 0,
                "Top": 0,
                "children": [
                {
                  "XamlType": "Microsoft.ReactNative.ViewPanel",
                  "Background": null,
                  "BorderBrush": "#00000000",
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Height": 26,
                  "HorizontalAlignment": "Stretch",
                  "Margin": "0,0,0,0",
                  "RenderSize": [800, 26],
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 800,
                  "Left": 0,
                  "Top": 12,
                  "children": [
                  {
                    "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                    "Clip": null,
                    "FlowDirection": "LeftToRight",
                    "Foreground": "#FF000000",
                    "Height": 26,
                    "HorizontalAlignment": "Stretch",
                    "Margin": "0,0,0,0",
                    "Padding": "0,0,0,0",
                    "RenderSize": [110, 26],
                    "Text": "Components",
                    "TextHighlighters": [],
                    "VerticalAlignment": "Stretch",
                    "Visibility": "Visible",
                    "Width": 111
                  }
                  ]
                }
                ]
              }
              ]
            },
            {
              "XamlType": "Microsoft.ReactNative.ViewPanel",
              "Background": "#FFF3F8FF",
              "BorderBrush": "#00000000",
              "BorderThickness": "0,0,0,0",
              "Clip": "Windows.UI.Xaml.Media.RectangleGeometry",
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "Height": 54,
              "HorizontalAlignment": "Stretch",
              "Margin": "0,0,0,0",
              "RenderSize": [800, 54],
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "Width": 800,
              "Left": 0,
              "Top": 56,
              "children": [
              {
                "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                "Clip": null,
                "FlowDirection": "LeftToRight",
                "Foreground": "#FF000000",
                "Height": 16,
                "HorizontalAlignment": "Stretch",
                "Margin": "0,0,0,0",
                "Padding": "0,0,0,0",
                "RenderSize": [170, 16],
                "Text": "LegacyTextBackgroundColorTest",
                "TextHighlighters": [],
                "VerticalAlignment": "Stretch",
                "Visibility": "Visible",
                "Width": 171
              },
              {
                "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                "Clip": null,
                "FlowDirection": "LeftToRight",
                "Foreground": "#FF000000",
                "Height": 26,
                "HorizontalAlignment": "Stretch",
                "Margin": "0,0,0,0",
                "Padding": "0,4,0,0",
                "RenderSize": [750, 26],
                "Text": "Legacy e2e test for Text with backgroundColor",
                "TextHighlighters": [],
                "VerticalAlignment": "Stretch",
                "Visibility": "Visible",
                "Width": 750
              }
              ]
            },
            {
              "XamlType": "Microsoft.ReactNative.ViewPanel",
              "Background": null,
              "BorderBrush": "#00000000",
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "Height": 490,
              "HorizontalAlignment": "Stretch",
              "Margin": "0,0,0,0",
              "RenderSize": [800, 490],
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "Width": 800,
              "Left": 0,
              "Top": 110,
              "children": [
              {
                "XamlType": "Microsoft.ReactNative.ViewPanel",
                "Background": null,
                "BorderBrush": "#00000000",
                "BorderThickness": "0,0,0,0",
                "Clip": null,
                "CornerRadius": "0,0,0,0",
                "FlowDirection": "LeftToRight",
                "Height": 95,
                "HorizontalAlignment": "Stretch",
                "Margin": "0,0,0,0",
                "RenderSize": [800, 94],
                "VerticalAlignment": "Stretch",
                "Visibility": "Visible",
                "Width": 800,
                "Left": 0,
                "Top": 0,
                "children": [
                {
                  "XamlType": "Microsoft.ReactNative.ViewPanel",
                  "Background": null,
                  "BorderBrush": "#00000000",
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Height": 95,
                  "HorizontalAlignment": "Stretch",
                  "Margin": "0,0,0,0",
                  "RenderSize": [800, 94],
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 800,
                  "Left": 0,
                  "Top": 0,
                  "AutomationId": "text-color-background-view",
                  "children": [
                  {
                    "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                    "Clip": null,
                    "FlowDirection": "LeftToRight",
                    "Foreground": "#FF000000",
                    "Height": 19,
                    "HorizontalAlignment": "Stretch",
                    "Margin": "0,0,0,0",
                    "Padding": "0,0,0,0",
                    "RenderSize": [800, 18],
                    "Text": "Outer no_color START_NESTED green DEEPER_NESTED blue END_NESTED attributes.",
                    "TextHighlighters": [{
"Background": "#FF008000",
"Ranges": [{"StartIndex":15,"Length":18}]
}
,{
"Background": "#FF008000",
"Ranges": [{"StartIndex":33,"Length":1}]
}
,{
"Background": "#FF0000FF",
"Ranges": [{"StartIndex":34,"Length":18}]
}
,{
"Background": "#FF008000",
"Ranges": [{"StartIndex":52,"Length":1}]
}
,{
"Background": "#FF008000",
"Ranges": [{"StartIndex":53,"Length":10}]
}
],
                    "VerticalAlignment": "Stretch",
                    "Visibility": "Visible",
                    "Width": 800
                  },
                  {
                    "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                    "Clip": null,
                    "FlowDirection": "LeftToRight",
                    "Foreground": "#FF000000",
                    "Height": 19,
                    "HorizontalAlignment": "Stretch",
                    "Margin": "0,0,0,0",
                    "Padding": "0,0,0,0",
                    "RenderSize": [800, 18],
                    "Text": "Outer no_color START_NESTED no_color DEEPER_NESTED blue END_NESTED attributes.",
                    "TextHighlighters": [{
"Background": "#FF0000FF",
"Ranges": [{"StartIndex":37,"Length":18}]
}
],
                    "VerticalAlignment": "Stretch",
                    "Visibility": "Visible",
                    "Width": 800
                  },
                  {
                    "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                    "Clip": null,
                    "FlowDirection": "LeftToRight",
                    "Foreground": "#FF000000",
                    "Height": 19,
                    "HorizontalAlignment": "Stretch",
                    "Margin": "0,0,0,0",
                    "Padding": "0,0,0,0",
                    "RenderSize": [800, 18],
                    "Text": "Outer no_color START_NESTED green DEEPER_NESTED inherit green END_NESTED attributes.",
                    "TextHighlighters": [{
"Background": "#FF008000",
"Ranges": [{"StartIndex":15,"Length":19}]
}
,{
"Background": "#FF008000",
"Ranges": [{"StartIndex":34,"Length":27}]
}
,{
"Background": "#FF008000",
"Ranges": [{"StartIndex":61,"Length":1}]
}
,{
"Background": "#FF008000",
"Ranges": [{"StartIndex":62,"Length":10}]
}
],
                    "VerticalAlignment": "Stretch",
                    "Visibility": "Visible",
                    "Width": 800
                  },
                  {
                    "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                    "Clip": null,
                    "FlowDirection": "LeftToRight",
                    "Foreground": "#FF000000",
                    "Height": 19,
                    "HorizontalAlignment": "Stretch",
                    "Margin": "0,0,0,0",
                    "Padding": "0,0,0,0",
                    "RenderSize": [800, 18],
                    "Text": "Outer red START_NESTED inherit red DEEPER_NESTED inherit red END_NESTED attributes.",
                    "TextHighlighters": [{
"Background": "#FFFF0000",
"Ranges": [{"StartIndex":0,"Length":9}]
}
,{
"Background": "#FFFF0000",
"Ranges": [{"StartIndex":9,"Length":1}]
}
,{
"Background": "#FFFF0000",
"Ranges": [{"StartIndex":10,"Length":25}]
}
,{
"Background": "#FFFF0000",
"Ranges": [{"StartIndex":35,"Length":25}]
}
,{
"Background": "#FFFF0000",
"Ranges": [{"StartIndex":60,"Length":1}]
}
,{
"Background": "#FFFF0000",
"Ranges": [{"StartIndex":61,"Length":10}]
}
,{
"Background": "#FFFF0000",
"Ranges": [{"StartIndex":71,"Length":1}]
}
,{
"Background": "#FFFF0000",
"Ranges": [{"StartIndex":72,"Length":11}]
}
],
                    "VerticalAlignment": "Stretch",
                    "Visibility": "Visible",
                    "Width": 800
                  },
                  {
                    "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                    "Clip": null,
                    "FlowDirection": "LeftToRight",
                    "Foreground": "#FF000000",
                    "Height": 19,
                    "HorizontalAlignment": "Stretch",
                    "Margin": "0,0,0,0",
                    "Padding": "0,0,0,0",
                    "RenderSize": [800, 18],
                    "Text": "Outer red START_NESTED green DEEPER_NESTED blue END_NESTED attributes.",
                    "TextHighlighters": [{
"Background": "#FFFF0000",
"Ranges": [{"StartIndex":0,"Length":9}]
}
,{
"Background": "#FFFF0000",
"Ranges": [{"StartIndex":9,"Length":1}]
}
,{
"Background": "#FF008000",
"Ranges": [{"StartIndex":10,"Length":18}]
}
,{
"Background": "#FF008000",
"Ranges": [{"StartIndex":28,"Length":1}]
}
,{
"Background": "#FF0000FF",
"Ranges": [{"StartIndex":29,"Length":18}]
}
,{
"Background": "#FF008000",
"Ranges": [{"StartIndex":47,"Length":1}]
}
,{
"Background": "#FF008000",
"Ranges": [{"StartIndex":48,"Length":10}]
}
,{
"Background": "#FFFF0000",
"Ranges": [{"StartIndex":58,"Length":1}]
}
,{
"Background": "#FFFF0000",
"Ranges": [{"StartIndex":59,"Length":11}]
}
],
                    "VerticalAlignment": "Stretch",
                    "Visibility": "Visible",
                    "Width": 800
                  }
                  ]
                }
                ]
              }
              ]
            }
            ]
          },
          {
            "XamlType": "Microsoft.ReactNative.ViewPanel",
            "Background": null,
            "BorderBrush": "#00000000",
            "BorderThickness": "0,0,0,0",
            "Clip": null,
            "CornerRadius": "0,0,0,0",
            "FlowDirection": "LeftToRight",
            "Height": 65,
            "HorizontalAlignment": "Stretch",
            "Margin": "0,0,0,0",
            "RenderSize": [800, 65],
            "VerticalAlignment": "Stretch",
            "Visibility": "Visible",
            "Width": 800,
            "Left": 0,
            "Top": 535,
            "children": [
            {
              "XamlType": "Microsoft.ReactNative.ViewPanel",
              "Background": null,
              "BorderBrush": "#00000000",
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "Height": 65,
              "HorizontalAlignment": "Stretch",
              "Margin": "0,0,0,0",
              "RenderSize": [800, 65],
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "Width": 800,
              "Left": 0,
              "Top": 0,
              "children": [
              {
                "XamlType": "Microsoft.ReactNative.ViewPanel",
                "Background": null,
                "BorderBrush": "#00000000",
                "BorderThickness": "0,0,0,0",
                "Clip": null,
                "CornerRadius": "0,0,0,0",
                "FlowDirection": "LeftToRight",
                "Height": 65,
                "HorizontalAlignment": "Stretch",
                "Margin": "0,0,0,0",
                "RenderSize": [800, 65],
                "VerticalAlignment": "Stretch",
                "Visibility": "Visible",
                "Width": 800,
                "Left": 0,
                "Top": 0,
                "children": [
                {
                  "XamlType": "Microsoft.ReactNative.ViewControl",
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "Height": 65,
                  "HorizontalAlignment": "Stretch",
                  "Margin": "0,0,0,0",
                  "Padding": "0,0,0,0",
                  "RenderSize": [266, 65],
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 267,
                  "AutomationId": "components-tab",
                  "children": [
                  {
                    "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                    "Background": null,
                    "BorderBrush": null,
                    "BorderThickness": "0,0,0,0",
                    "Clip": null,
                    "CornerRadius": "0,0,0,0",
                    "FlowDirection": "LeftToRight",
                    "Foreground": "#FF000000",
                    "HorizontalAlignment": "Stretch",
                    "Margin": "0,0,0,0",
                    "Padding": "0,0,0,0",
                    "RenderSize": [266, 65],
                    "VerticalAlignment": "Stretch",
                    "Visibility": "Visible",
                    "children": [
                    {
                      "XamlType": "Microsoft.ReactNative.ViewPanel",
                      "Background": "#FFF3F8FF",
                      "BorderBrush": "#00000000",
                      "BorderThickness": "0,0,0,0",
                      "Clip": null,
                      "CornerRadius": "0,0,0,0",
                      "FlowDirection": "LeftToRight",
                      "Height": 65,
                      "HorizontalAlignment": "Stretch",
                      "Margin": "0,0,0,0",
                      "RenderSize": [266, 65],
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 267,
                      "Left": 0,
                      "Top": 0,
                      "children": [
                      {
                        "XamlType": "Microsoft.ReactNative.ViewPanel",
                        "Background": null,
                        "BorderBrush": "#00000000",
                        "BorderThickness": "0,2,0,0",
                        "Clip": null,
                        "CornerRadius": "0,0,0,0",
                        "FlowDirection": "LeftToRight",
                        "Height": 65,
                        "HorizontalAlignment": "Stretch",
                        "Margin": "0,0,0,0",
                        "RenderSize": [266, 65],
                        "VerticalAlignment": "Stretch",
                        "Visibility": "Visible",
                        "Width": 267,
                        "Left": 0,
                        "Top": 0,
                        "children": [
                        {
                          "XamlType": "Windows.UI.Xaml.DependencyObject"
                        },
                        {
                          "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                          "Clip": null,
                          "FlowDirection": "LeftToRight",
                          "Foreground": "#FFB1B4BA",
                          "Height": 19,
                          "HorizontalAlignment": "Stretch",
                          "Margin": "0,0,0,0",
                          "Padding": "0,0,0,0",
                          "RenderSize": [80, 18],
                          "Text": "Components",
                          "TextHighlighters": [],
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "Width": 80
                        }
                        ]
                      }
                      ]
                    }
                    ]
                  }
                  ]
                },
                {
                  "XamlType": "Microsoft.ReactNative.ViewPanel",
                  "Background": null,
                  "BorderBrush": "#00000000",
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Height": 65,
                  "HorizontalAlignment": "Stretch",
                  "Margin": "0,0,0,0",
                  "RenderSize": [266, 65],
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 266,
                  "Left": 267,
                  "Top": 0,
                  "children": [
                  {
                    "XamlType": "Windows.UI.Xaml.DependencyObject"
                  },
                  {
                    "XamlType": "Microsoft.ReactNative.ViewPanel",
                    "Background": null,
                    "BorderBrush": "#00000000",
                    "BorderThickness": "0,0,0,0",
                    "Clip": null,
                    "CornerRadius": "0,0,0,0",
                    "FlowDirection": "LeftToRight",
                    "Height": 65,
                    "HorizontalAlignment": "Stretch",
                    "Margin": "0,0,0,0",
                    "RenderSize": [266, 65],
                    "VerticalAlignment": "Stretch",
                    "Visibility": "Visible",
                    "Width": 266,
                    "Left": 0,
                    "Top": 0,
                    "children": [
                    {
                      "XamlType": "Microsoft.ReactNative.ViewControl",
                      "Background": null,
                      "BorderBrush": null,
                      "BorderThickness": "0,0,0,0",
                      "Clip": null,
                      "CornerRadius": "0,0,0,0",
                      "FlowDirection": "LeftToRight",
                      "Foreground": "#FF000000",
                      "Height": 50,
                      "HorizontalAlignment": "Stretch",
                      "Margin": "0,0,0,0",
                      "Padding": "0,0,0,0",
                      "RenderSize": [50, 50],
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 50,
                      "AutomationId": "bookmarks-tab",
                      "children": [
                      {
                        "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                        "Background": null,
                        "BorderBrush": null,
                        "BorderThickness": "0,0,0,0",
                        "Clip": null,
                        "CornerRadius": "0,0,0,0",
                        "FlowDirection": "LeftToRight",
                        "Foreground": "#FF000000",
                        "HorizontalAlignment": "Stretch",
                        "Margin": "0,0,0,0",
                        "Padding": "0,0,0,0",
                        "RenderSize": [50, 50],
                        "VerticalAlignment": "Stretch",
                        "Visibility": "Visible",
                        "children": [
                        {
                          "XamlType": "Microsoft.ReactNative.ViewPanel",
                          "Background": "#00FFFFFF",
                          "BorderBrush": "#00000000",
                          "BorderThickness": "0,0,0,0",
                          "Clip": null,
                          "CornerRadius": "0,0,0,0",
                          "FlowDirection": "LeftToRight",
                          "Height": 50,
                          "HorizontalAlignment": "Stretch",
                          "Margin": "0,0,0,0",
                          "RenderSize": [50, 50],
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "Width": 50,
                          "Left": 0,
                          "Top": 0,
                          "children": [
                          {
                            "XamlType": "Windows.UI.Xaml.Controls.Border",
                            "Background": "#FF005DFF",
                            "BorderBrush": null,
                            "BorderThickness": "0,0,0,0",
                            "Clip": null,
                            "CornerRadius": "500,500,500,500",
                            "FlowDirection": "LeftToRight",
                            "Height": 50,
                            "HorizontalAlignment": "Stretch",
                            "Margin": "0,0,0,0",
                            "Padding": "0,0,0,0",
                            "RenderSize": [50, 50],
                            "VerticalAlignment": "Stretch",
                            "Visibility": "Visible",
                            "Width": 50,
                            "children": [
                            {
                              "XamlType": "Microsoft.ReactNative.ViewPanel",
                              "Background": null,
                              "BorderBrush": "#00000000",
                              "BorderThickness": "0,0,0,0",
                              "Clip": null,
                              "CornerRadius": "500,500,500,500",
                              "FlowDirection": "LeftToRight",
                              "HorizontalAlignment": "Stretch",
                              "Margin": "0,0,0,0",
                              "RenderSize": [50, 50],
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "Left": 0,
                              "Top": 0,
                              "children": [
                              {
                                "XamlType": "Windows.UI.Xaml.DependencyObject"
                              }
                              ]
                            }
                            ]
                          }
                          ]
                        }
                        ]
                      }
                      ]
                    }
                    ]
                  }
                  ]
                },
                {
                  "XamlType": "Microsoft.ReactNative.ViewControl",
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "Height": 65,
                  "HorizontalAlignment": "Stretch",
                  "Margin": "0,0,0,0",
                  "Padding": "0,0,0,0",
                  "RenderSize": [266, 65],
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 267,
                  "AutomationId": "apis-tab",
                  "children": [
                  {
                    "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                    "Background": null,
                    "BorderBrush": null,
                    "BorderThickness": "0,0,0,0",
                    "Clip": null,
                    "CornerRadius": "0,0,0,0",
                    "FlowDirection": "LeftToRight",
                    "Foreground": "#FF000000",
                    "HorizontalAlignment": "Stretch",
                    "Margin": "0,0,0,0",
                    "Padding": "0,0,0,0",
                    "RenderSize": [266, 65],
                    "VerticalAlignment": "Stretch",
                    "Visibility": "Visible",
                    "children": [
                    {
                      "XamlType": "Microsoft.ReactNative.ViewPanel",
                      "Background": "#FFF3F8FF",
                      "BorderBrush": "#00000000",
                      "BorderThickness": "0,0,0,0",
                      "Clip": null,
                      "CornerRadius": "0,0,0,0",
                      "FlowDirection": "LeftToRight",
                      "Height": 65,
                      "HorizontalAlignment": "Stretch",
                      "Margin": "0,0,0,0",
                      "RenderSize": [266, 65],
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 267,
                      "Left": 0,
                      "Top": 0,
                      "children": [
                      {
                        "XamlType": "Microsoft.ReactNative.ViewPanel",
                        "Background": null,
                        "BorderBrush": "#00000000",
                        "BorderThickness": "0,0,0,0",
                        "Clip": null,
                        "CornerRadius": "0,0,0,0",
                        "FlowDirection": "LeftToRight",
                        "Height": 65,
                        "HorizontalAlignment": "Stretch",
                        "Margin": "0,0,0,0",
                        "RenderSize": [266, 65],
                        "VerticalAlignment": "Stretch",
                        "Visibility": "Visible",
                        "Width": 267,
                        "Left": 0,
                        "Top": 0,
                        "children": [
                        {
                          "XamlType": "Windows.UI.Xaml.DependencyObject"
                        },
                        {
                          "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                          "Clip": null,
                          "FlowDirection": "LeftToRight",
                          "Foreground": "#FFB1B4BA",
                          "Height": 19,
                          "HorizontalAlignment": "Stretch",
                          "Margin": "0,0,0,0",
                          "Padding": "0,0,0,0",
                          "RenderSize": [26, 18],
                          "Text": "APIs",
                          "TextHighlighters": [],
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "Width": 27
                        }
                        ]
                      }
                      ]
                    }
                    ]
                  }
                  ]
                }
                ]
              }
              ]
            }
            ]
          }
          ]
        },
        {
          "XamlType": "Microsoft.ReactNative.ViewPanel",
          "Background": null,
          "BorderBrush": "#00000000",
          "BorderThickness": "0,0,0,0",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "Height": 53,
          "HorizontalAlignment": "Stretch",
          "Margin": "0,0,0,0",
          "RenderSize": [780, 53],
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "Width": 780,
          "Left": 10,
          "Top": 527,
          "children": [
          {
            "XamlType": "Windows.UI.Xaml.Controls.Border",
            "Background": null,
            "BorderBrush": null,
            "BorderThickness": "0,0,0,0",
            "Clip": null,
            "CornerRadius": "8,8,8,8",
            "FlowDirection": "LeftToRight",
            "Height": 48,
            "HorizontalAlignment": "Stretch",
            "Margin": "0,0,0,0",
            "Padding": "0,0,0,0",
            "RenderSize": [780, 48],
            "VerticalAlignment": "Stretch",
            "Visibility": "Visible",
            "Width": 780,
            "children": [
            {
              "XamlType": "Microsoft.ReactNative.ViewPanel",
              "Background": null,
              "BorderBrush": "#00000000",
              "BorderThickness": "0,0,0,0",
              "Clip": "Windows.UI.Xaml.Media.RectangleGeometry",
              "CornerRadius": "8,8,8,8",
              "FlowDirection": "LeftToRight",
              "HorizontalAlignment": "Stretch",
              "Margin": "0,0,0,0",
              "RenderSize": [780, 48],
              "VerticalAlignment": "Stretch",
              "Visibility": "Visible",
              "Left": 0,
              "Top": 0,
              "children": [
              {
                "XamlType": "Microsoft.ReactNative.ViewPanel",
                "Background": "#FFFFFFFF",
                "BorderBrush": "#00000000",
                "BorderThickness": "0,0,0,0",
                "Clip": null,
                "CornerRadius": "0,0,0,0",
                "FlowDirection": "LeftToRight",
                "Height": 48,
                "HorizontalAlignment": "Stretch",
                "Margin": "0,0,0,0",
                "RenderSize": [780, 48],
                "VerticalAlignment": "Stretch",
                "Visibility": "Visible",
                "Width": 780,
                "Left": 0,
                "Top": 1,
                "children": [
                {
                  "XamlType": "Microsoft.ReactNative.ViewControl",
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "Height": 48,
                  "HorizontalAlignment": "Stretch",
                  "Margin": "0,0,0,0",
                  "Padding": "0,0,0,0",
                  "RenderSize": [780, 48],
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 780,
                  "children": [
                  {
                    "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                    "Background": null,
                    "BorderBrush": null,
                    "BorderThickness": "0,0,0,0",
                    "Clip": null,
                    "CornerRadius": "0,0,0,0",
                    "FlowDirection": "LeftToRight",
                    "Foreground": "#FF000000",
                    "HorizontalAlignment": "Stretch",
                    "Margin": "0,0,0,0",
                    "Padding": "0,0,0,0",
                    "RenderSize": [780, 48],
                    "VerticalAlignment": "Stretch",
                    "Visibility": "Visible",
                    "children": [
                    {
                      "XamlType": "Microsoft.ReactNative.ViewPanel",
                      "Background": "#FF333333",
                      "BorderBrush": "#00000000",
                      "BorderThickness": "0,0,0,0",
                      "Clip": null,
                      "CornerRadius": "0,0,0,0",
                      "FlowDirection": "LeftToRight",
                      "Height": 48,
                      "HorizontalAlignment": "Stretch",
                      "Margin": "0,0,0,0",
                      "RenderSize": [780, 48],
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 780,
                      "Left": 0,
                      "Top": 0,
                      "children": [
                      {
                        "XamlType": "Windows.UI.Xaml.Controls.Border",
                        "Background": null,
                        "BorderBrush": null,
                        "BorderThickness": "0,0,0,0",
                        "Clip": null,
                        "CornerRadius": "8,8,8,8",
                        "FlowDirection": "LeftToRight",
                        "Height": 23,
                        "HorizontalAlignment": "Stretch",
                        "Margin": "0,0,0,0",
                        "Padding": "0,0,0,0",
                        "RenderSize": [756, 22],
                        "VerticalAlignment": "Stretch",
                        "Visibility": "Visible",
                        "Width": 756,
                        "children": [
                        {
                          "XamlType": "Microsoft.ReactNative.ViewPanel",
                          "Background": null,
                          "BorderBrush": "#00000000",
                          "BorderThickness": "0,0,0,0",
                          "Clip": null,
                          "CornerRadius": "8,8,8,8",
                          "FlowDirection": "LeftToRight",
                          "HorizontalAlignment": "Stretch",
                          "Margin": "0,0,0,0",
                          "RenderSize": [756, 22],
                          "VerticalAlignment": "Stretch",
                          "Visibility": "Visible",
                          "Left": 0,
                          "Top": 0,
                          "children": [
                          {
                            "XamlType": "Windows.UI.Xaml.Controls.Border",
                            "Background": "#FFFFFFFF",
                            "BorderBrush": null,
                            "BorderThickness": "0,0,0,0",
                            "Clip": null,
                            "CornerRadius": "25,25,25,25",
                            "FlowDirection": "LeftToRight",
                            "Height": 23,
                            "HorizontalAlignment": "Stretch",
                            "Margin": "0,0,0,0",
                            "Padding": "0,0,0,0",
                            "RenderSize": [22, 22],
                            "VerticalAlignment": "Stretch",
                            "Visibility": "Visible",
                            "Width": 22,
                            "children": [
                            {
                              "XamlType": "Microsoft.ReactNative.ViewPanel",
                              "Background": null,
                              "BorderBrush": "#00000000",
                              "BorderThickness": "0,0,0,0",
                              "Clip": null,
                              "CornerRadius": "25,25,25,25",
                              "FlowDirection": "LeftToRight",
                              "HorizontalAlignment": "Stretch",
                              "Margin": "0,0,0,0",
                              "RenderSize": [22, 22],
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "Left": 0,
                              "Top": 0,
                              "children": [
                              {
                                "XamlType": "Windows.UI.Xaml.Controls.Border",
                                "Background": "#FFF35369",
                                "BorderBrush": null,
                                "BorderThickness": "0,0,0,0",
                                "Clip": null,
                                "CornerRadius": "25,25,25,25",
                                "FlowDirection": "LeftToRight",
                                "Height": 19,
                                "HorizontalAlignment": "Stretch",
                                "Margin": "0,0,0,0",
                                "Padding": "0,0,0,0",
                                "RenderSize": [18, 18],
                                "VerticalAlignment": "Stretch",
                                "Visibility": "Visible",
                                "Width": 18,
                                "children": [
                                {
                                  "XamlType": "Microsoft.ReactNative.ViewPanel",
                                  "Background": null,
                                  "BorderBrush": "#00000000",
                                  "BorderThickness": "0,0,0,0",
                                  "Clip": null,
                                  "CornerRadius": "25,25,25,25",
                                  "FlowDirection": "LeftToRight",
                                  "HorizontalAlignment": "Stretch",
                                  "Margin": "0,0,0,0",
                                  "RenderSize": [18, 18],
                                  "VerticalAlignment": "Stretch",
                                  "Visibility": "Visible",
                                  "Left": 0,
                                  "Top": 0,
                                  "children": [
                                  {
                                    "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                                    "Clip": null,
                                    "FlowDirection": "LeftToRight",
                                    "Foreground": "#FFFFFFFF",
                                    "Height": 19,
                                    "HorizontalAlignment": "Stretch",
                                    "Margin": "0,0,0,0",
                                    "Padding": "0,0,0,0",
                                    "RenderSize": [10, 18],
                                    "Text": "3",
                                    "TextHighlighters": [],
                                    "VerticalAlignment": "Stretch",
                                    "Visibility": "Visible",
                                    "Width": 10
                                  }
                                  ]
                                }
                                ]
                              }
                              ]
                            }
                            ]
                          },
                          {
                            "XamlType": "Microsoft.ReactNative.ViewPanel",
                            "Background": null,
                            "BorderBrush": "#00000000",
                            "BorderThickness": "1,0,0,0",
                            "Clip": null,
                            "CornerRadius": "0,0,0,0",
                            "FlowDirection": "LeftToRight",
                            "Height": 23,
                            "HorizontalAlignment": "Stretch",
                            "Margin": "0,0,0,0",
                            "RenderSize": [701, 22],
                            "VerticalAlignment": "Stretch",
                            "Visibility": "Visible",
                            "Width": 701,
                            "Left": 30,
                            "Top": 0,
                            "children": [
                            {
                              "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
                              "Clip": null,
                              "FlowDirection": "LeftToRight",
                              "Foreground": "#FFFFFFFF",
                              "Height": 23,
                              "HorizontalAlignment": "Stretch",
                              "Margin": "0,0,0,0",
                              "Padding": "0,0,0,0",
                              "RenderSize": [692, 22],
                              "Text": "NativeUIManager.getConstantsForViewManager('ForwardRef(TextInput)') threw an exception. Invariant Violation: Calling synchronous methods on native modules is not supported in Chrome.\n\n Consider providing alternative methods to expose this method in debug mode, e.g. by exposing constants ahead-of-time. ",
                              "TextHighlighters": [],
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "Width": 692
                            }
                            ]
                          },
                          {
                            "XamlType": "Microsoft.ReactNative.ViewPanel",
                            "Background": null,
                            "BorderBrush": "#00000000",
                            "BorderThickness": "0,0,0,0",
                            "Clip": null,
                            "CornerRadius": "0,0,0,0",
                            "FlowDirection": "LeftToRight",
                            "Height": 20,
                            "HorizontalAlignment": "Stretch",
                            "Margin": "0,0,0,0",
                            "RenderSize": [20, 20],
                            "VerticalAlignment": "Stretch",
                            "Visibility": "Visible",
                            "Width": 20,
                            "Left": 736,
                            "Top": 2,
                            "children": [
                            {
                              "XamlType": "Microsoft.ReactNative.ViewControl",
                              "Background": null,
                              "BorderBrush": null,
                              "BorderThickness": "0,0,0,0",
                              "Clip": null,
                              "CornerRadius": "0,0,0,0",
                              "FlowDirection": "LeftToRight",
                              "Foreground": "#FF000000",
                              "Height": 20,
                              "HorizontalAlignment": "Stretch",
                              "Margin": "0,0,0,0",
                              "Padding": "0,0,0,0",
                              "RenderSize": [20, 20],
                              "VerticalAlignment": "Stretch",
                              "Visibility": "Visible",
                              "Width": 20,
                              "children": [
                              {
                                "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                                "Background": null,
                                "BorderBrush": null,
                                "BorderThickness": "0,0,0,0",
                                "Clip": null,
                                "CornerRadius": "0,0,0,0",
                                "FlowDirection": "LeftToRight",
                                "Foreground": "#FF000000",
                                "HorizontalAlignment": "Stretch",
                                "Margin": "0,0,0,0",
                                "Padding": "0,0,0,0",
                                "RenderSize": [20, 20],
                                "VerticalAlignment": "Stretch",
                                "Visibility": "Visible",
                                "children": [
                                {
                                  "XamlType": "Windows.UI.Xaml.Controls.Border",
                                  "Background": "#4DFFFFFF",
                                  "BorderBrush": null,
                                  "BorderThickness": "0,0,0,0",
                                  "Clip": null,
                                  "CornerRadius": "25,25,25,25",
                                  "FlowDirection": "LeftToRight",
                                  "HorizontalAlignment": "Stretch",
                                  "Margin": "0,0,0,0",
                                  "Padding": "0,0,0,0",
                                  "RenderSize": [20, 20],
                                  "VerticalAlignment": "Stretch",
                                  "Visibility": "Visible",
                                  "children": [
                                  {
                                    "XamlType": "Microsoft.ReactNative.ViewPanel",
                                    "Background": null,
                                    "BorderBrush": "#00000000",
                                    "BorderThickness": "0,0,0,0",
                                    "Clip": null,
                                    "CornerRadius": "25,25,25,25",
                                    "FlowDirection": "LeftToRight",
                                    "Height": 20,
                                    "HorizontalAlignment": "Stretch",
                                    "Margin": "0,0,0,0",
                                    "RenderSize": [20, 20],
                                    "VerticalAlignment": "Stretch",
                                    "Visibility": "Visible",
                                    "Width": 20,
                                    "Left": 0,
                                    "Top": 0,
                                    "children": [
                                    {
                                      "XamlType": "Windows.UI.Xaml.DependencyObject"
                                    }
                                    ]
                                  }
                                  ]
                                }
                                ]
                              }
                              ]
                            }
                            ]
                          }
                          ]
                        }
                        ]
                      }
                      ]
                    }
                    ]
                  }
                  ]
                }
                ]
              }
              ]
            }
            ]
          }
          ]
        }
        ]
      }
      ]
    }
    ]
  }
  ]
}
```
</detail>
snippet:
![image](https://user-images.githubusercontent.com/22989529/110055775-6efeae80-7d12-11eb-9331-6e2afda9ae15.png)

So maybe the element you are selecting out of the tree just happens to be at 0, 0?

NickGerleman commented:
> It looks like Left/Top is non-zero for some elements, here's the dump I got:
> 
> So maybe the element you are selecting out of the tree just happens to be at 0, 0?

Just to clarify, we are not looking for "Top/Left" on ViewPanel, but should be looking for "ViewPanel.Left" and "ViewPanel.Top" on everything created by a ViewManager.

It would be worth spinning up the live visual tree to take a look at these.

NickGerleman commented:
Some more context in the thread for this old layout bug https://github.com/microsoft/react-native-windows/issues/6586#issuecomment-731404831

NickGerleman commented:
During layout of ShadowNodes, we set ViewPanel.Left, ViewPanel.Top, etc with outputs from Yoga layout. XAML then positions the elements using custom arrange logic, derived from these elements.

The flow looks like:
1.Yoga nodes are created for each ShadowNode in `NativeUIManager::DoLayout`
1. Layout props set per-shadownode in `NativeUIManager::DoLayout`
1. Shadownode XAML element props set in `ViewManagerBase::SetLayoutProps`

NickGerleman commented:
The above flow can be useful to know and look through for any layout/positioning related tasks. Debugging it was able to help inform my own understanding of the RNW layout and positioning flow.

asklar commented:
re: ViewPanel.Left on non-ViewPanel types: this is correct. I think we may need to change mainpage.xaml.cs to instead pass Top and Left in the last param of the call to dump the visual tree, something like ` new AttachedProperty[] { new AttachedProperty() { Name ="Top", Property = Microsoft.ReactNative.ViewPanel.TopProperty }, /* same for Left */ }`

rectified95 commented:
@asklar I changed it as you suggested - now passing the props in MainPage.xaml.cs, instead of as the `additionalProperties` parameter in the JS test files.
@NickGerleman I've updated the snapshots - there're still many zero entries, but I can see more of non-zero ones.

rectified95 commented:
Btw, @asklar the JSON snippet you posted above, where `Left==736` seems to come from a redbox that's below the actual test.
Other than that, I'm not seeing the value of left being different from zero anywhere - still could be by chance.
